internal/webhooks/clusterclass.go

```diff
@@ -99,6 +103,21 @@ func (webhook *ClusterClass) ValidateUpdate(_ context.Context, oldObj, newObj ru
 
 // ValidateDelete implements validation for ClusterClass delete.
 func (webhook *ClusterClass) ValidateDelete(ctx context.Context, obj runtime.Object) error {
+	clusterClass, ok := obj.(*clusterv1.ClusterClass)
+	if !ok {
+		return apierrors.NewBadRequest(fmt.Sprintf("expected a ClusterClass but got a %T", obj))
+	}
+
+	clusters, err := webhook.getClustersUsingClusterClass(clusterClass)
+	if err != nil {
+		return apierrors.NewInvalid(clusterv1.GroupVersion.WithKind("ClusterClass").GroupKind(), clusterClass.Name, err)
+	}
+
+	if len(clusters) != 0 {
+		// TODO(killianmuldoon): Improve error here to include the names of some clusters using the clusterClass.
+		return field.Forbidden(field.NewPath("clusterClass"),
```

Comment:
I think this also should use ~  `apierrors.NewInvalid(clusterv1.GroupVersion.WithKind("ClusterClass").GroupKind(), new.Name, allErrs)` (I missed this the first time)

Reply:
Why use invalid rather than forbidden here?